h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1808,12 +1807,13 @@ private void compactMoveChunks(Iterable<Chunk> move) {
                 // store a new chunk with updated metadata (hopefully within a file)
                 store(originalBlokCount, postEvacuationBlockCount);
                 sync();
-                // and if last chunk did not fit within a file, since we can use
-                // previously reserved area [originalBlokCount, postEvacuationBlockCount) now,
-                // try to move it closer to BOF
-                moved = !moved && moveChunkInside(chunkToMove, postEvacuationBlockCount);
-                if (moveChunkInside(lastChunk, postEvacuationBlockCount) || moved) {
-                    store(postEvacuationBlockCount, -1);
+                // if chunkToMove did not fit within originalBlockCount (move is false), and since now
+                // previously reserved area [originalBlokCount, postEvacuationBlockCount) also can be used,
+                // lets try to move that chunk into this area, closer to BOF
+                long lastBoundary = moved ? postEvacuationBlockCount : chunkToMove.block;
```

Comment:
In line 1813 there is a small problem. If the chunkToMove hasn't been moved and was [0, leftmostBlock), the lastBoundary is too small and incorrect.

Reply:
>In line 1813 there is a small problem. If the chunkToMove was stored into [0, leftmostBlock) in line 1785 and hasn't been moved, then the lastBoundary is too small and incorrect.

My bad. I wrongly assumed that **movedToEOF** will always be false in that scenario. Need to be fixed.

>And if the chunkToMove has been moved in line 1805 and the lastChunk isn't moved in line 1815, we shouldn't call store() in line 1816, because of since last store() in line 1808, no any chunk has been moved.

In this scenario **move** variable will flip from true to false on 1814, and if will be skipped.